src/StackManager.js

```diff
@@ -63,30 +67,35 @@ export default class StackManager {
   makeOutputDir() {
     return fs.ensureDir(buildPath(this.getProjectPath()));
   }
-  execDocker(...args) {
-    const command = clean(args).join(' ');
-
-    return exec(`docker ${command}`);
-  }
-  execDockerCompose(...args) {
-    const file = dockerComposePath(this.getProjectPath());
-    const command = clean(args).join(' ');
-
-    return exec(`docker-compose --file ${file} ${command}`);
+  execDocker(subcommand, args) {
+    return exec('docker', [
+      subcommand,
+      ...args,
+    ]);
   }
-  spawnDockerCompose(...args) {
+  spawnDockerCompose(subcommand, args) {
     const file = dockerComposePath(this.getProjectPath());
-    const child = spawn('docker-compose', clean(['--file', file, ...args]), { stdio: 'inherit' });
+    const child = spawn('docker-compose', [
+      '--file', file,
+      '--project-name', this.getProjectName(),
+      subcommand,
+      ...args,
+    ], { stdio: 'inherit' });
 
     // cancel default SIGINT behaviour
     process.on('SIGINT', () => {});
 
     return child;
   }
-  async listContainers(serviceName = null) {
-    const stdout = await this.execDockerCompose('ps -q', serviceName);
+  async listRunningContainers(serviceName) {
+    const label = `${this.getProjectName()}.${serviceName}`;
+    const stdout = await this.execDocker('ps', clean([
+      '-q',
```

Comment:
since line width is not a concern, let's have full option names for better readability